src/Analyzer.Cli.FunctionalTests/CommandLineParserTests.cs

```diff
@@ -110,7 +126,26 @@ public void AnalyzeDirectory_ValidInputValues_AnalyzesExpectedNumberOfFiles()
             var result = _commandLineParser.InvokeCommandLineAPIAsync(args);
 
             Assert.AreEqual((int)ExitCode.ErrorAndViolation, result.Result);
-            StringAssert.Contains(outputWriter.ToString(), "Analyzed 10 files");
+            StringAssert.Contains(outputWriter.ToString(), "Analyzed 15 files");
+        }
+
+        [TestMethod]
+        public void AnalyzeDirectory_ValidInputValues_AnalyzesExpectedNumberOfFilesWithAutoDetectedParameters()
+        {
+            var args = new string[] { "analyze-directory", Path.Combine(Directory.GetCurrentDirectory(), "Tests", "ToTestSeparateParametersFile") };
+
+            using StringWriter outputWriter = new();
+            Console.SetOut(outputWriter);
+
+            var result = _commandLineParser.InvokeCommandLineAPIAsync(args);
+
+            Assert.AreEqual((int)ExitCode.Success, result.Result);
+            StringAssert.Contains(outputWriter.ToString(), "Analyzed 5 files");
```

Comment:
Can you also validate that the template is indeed analyzed only 5 times? For example, counting how many times the template file path shows in the output

Reply:
Added assertions for this, see [`881b3ad` (#297)](https://github.com/Azure/template-analyzer/pull/297/commits/881b3ada7810321ca1ab57306b974464a531c546)